aspnetcore/blazor/security/index.md

```diff
@@ -0,0 +1,486 @@
+
+The command creates a folder named with the value in the `{APP NAME}` placeholder and uses the same value as the app name. For more information, see the [dotnet new](/dotnet/core/tools/dotnet-new) command in the .NET Core Guide.
+
+---
+
+### Blazor client-side authentication
+
+In Blazor client-side apps, authentication checks can be bypassed because all client-side code can be modified by users. The same is true for all client-side app technologies, including JavaScript SPA frameworks or native apps for any operating system.
+
+Implement an [AuthenticationStateProvider service](#authenticationstateprovider-service) to set up authentication in a Blazor client-side app. For more information, see the [Implement a custom AuthenticationStateProvider](#implement-a-custom-authenticationstateprovider) section.
+
+### AuthenticationStateProvider service
+
+Blazor server-side apps include a built-in `AuthenticationStateProvider` service that obtains authentication state data from ASP.NET Core's `HttpContext.User`. This is how authentication state integrates with existing ASP.NET Core server-side authentication mechanisms.
+
+`AuthenticationStateProvider` is the underlying service used by the Authorize View component (`AuthorizeView`) and Cascading Authentication State component (`<CascadingAuthenticationState>`) to get the authentication state.
+
+You don't typically use `AuthenticationStateProvider` directly. Use the [Authorize View component](#authorize-view-component) or [Task<AuthenticationState>](#expose-the-authentication-state-as-a-cascading-parameter) approaches described later in this article. The main drawback to using `AuthenticationStateProvider` directly is that the component isn't notified automatically if the underlying authentication state data changes.
+
+The `AuthenticationStateProvider` service can provide the current user's <xref:System.Security.Claims.ClaimsPrincipal> data, as shown in the following example:
+
+```cshtml
+@page "/"
+@inject AuthenticationStateProvider AuthenticationStateProvider
+
+<button @onclick="@LogUsername">Write user info to console</button>
+
+@code {
+    private async Task LogUsername()
+    {
+        var authState = await AuthenticationStateProvider.GetAuthenticationStateAsync();
+        var user = authState.User;
+
+        if (user.Identity.IsAuthenticated)
+        {
+            Console.WriteLine($"{user.Identity.Name} is authenticated.");
+        }
+        else
+        {
+            Console.WriteLine("The user is NOT authenticated.");
+        }
+    }
+}
+```
+
+If `user.Identity.IsAuthenticated` is `true` and because the user is a <xref:System.Security.Claims.ClaimsPrincipal>, claims can be enumerated and membership in roles evaluated.
+
+For more information on dependency injection (DI) and services, see <xref:blazor/dependency-injection> and <xref:fundamentals/dependency-injection>.
+
+### Implement a custom AuthenticationStateProvider
+
+> [!WARNING]
```

Comment:
I think we should remove this note and the following paragraph at this point. We're already planning on putting a custom AuthenticationStateProvider into the Blazor server-side template to deal with authentication revalidation.

Reply:
+1 to this. I came here to leave this exact same comment.